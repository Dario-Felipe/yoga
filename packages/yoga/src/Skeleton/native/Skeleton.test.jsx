import React from 'react';
import { render } from '@testing-library/react-native';

import { ThemeProvider, Skeleton } from '../..';

describe('<Skeleton />', () => {
  it('should render the circular skeleton', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Skeleton type="circular" width={64} height={64} />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the rectangular skeleton', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Skeleton type="rectangular" width={400} height={200} />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the text skeleton', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Skeleton type="text" variant="body1" width={61} height={32} />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render all text skeleton variants correctly', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Skeleton type="text" variant="h1" width="100" />
        <Skeleton type="text" variant="h2" width="100" />
        <Skeleton type="text" variant="h3" width="100" />
        <Skeleton type="text" variant="h4" width="100" />
        <Skeleton type="text" variant="h5" width="100" />
        <Skeleton type="text" variant="body1" width="100" />
        <Skeleton type="text" variant="body2" width="100" />
        <Skeleton type="text" variant="overline" width="100" />
        <Skeleton type="text" variant="exception" width="100" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
