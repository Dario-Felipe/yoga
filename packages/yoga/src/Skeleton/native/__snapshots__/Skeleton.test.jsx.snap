// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> should render all background color variants correctly 1`] = `
[
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 24,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="body1"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="secondary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FAF9F5",
            "display": "flex",
            "height": 24,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="body1"
      width="100"
    />
  </View>,
]
`;

exports[`<Skeleton /> should render all text skeleton variants correctly 1`] = `
[
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 56,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="h1"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 48,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="h2"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 40,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="h3"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 32,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="h4"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 28,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="h5"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 24,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="body1"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 20,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="body2"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 16,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="overline"
      width="100"
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "opacity": 0.4,
      }
    }
  >
    <View
      color="primary"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F5F5FA",
            "display": "flex",
            "height": 12,
            "width": 100,
          },
        ]
      }
      type="text"
      variant="exception"
      width="100"
    />
  </View>,
]
`;

exports[`<Skeleton /> should render the circular skeleton 1`] = `
<View
  collapsable={false}
  style={
    {
      "opacity": 0.4,
    }
  }
>
  <View
    color="primary"
    height={64}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F5F5FA",
          "borderRadius": 9999,
          "display": "flex",
          "height": 64,
          "width": 64,
        },
      ]
    }
    type="circular"
    width={64}
  />
</View>
`;

exports[`<Skeleton /> should render the rectangular skeleton 1`] = `
<View
  collapsable={false}
  style={
    {
      "opacity": 0.4,
    }
  }
>
  <View
    color="primary"
    height={200}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F5F5FA",
          "display": "flex",
          "height": 200,
          "width": 400,
        },
      ]
    }
    type="rectangular"
    width={400}
  />
</View>
`;

exports[`<Skeleton /> should render the text skeleton 1`] = `
<View
  collapsable={false}
  style={
    {
      "opacity": 0.4,
    }
  }
>
  <View
    color="primary"
    height={32}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F5F5FA",
          "display": "flex",
          "height": 24,
          "width": 61,
        },
      ]
    }
    type="text"
    variant="body1"
    width={61}
  />
</View>
`;
